tests/ssl-opt.sh

```diff
@@ -1695,23 +1775,35 @@ run_test() {
         esac
     fi
 
-    # does this test use a proxy?
+    # Does this test specify a proxy?
     if [ "X$1" = "X-p" ]; then
         PXY_CMD="$2"
         shift 2
     else
         PXY_CMD=""
     fi
 
+    # Does this test force a specific port?
+    if [ "$1" = "-P" ]; then
+        THIS_SRV_PORT="$2"
+        shift 2
+    else
+        THIS_SRV_PORT="$SRV_PORT"
+    fi
+
     # get commands and client output
     SRV_CMD="$1"
     CLI_CMD="$2"
     CLI_EXPECT="$3"
     shift 3
 
-    # Check if test uses files
-    case "$SRV_CMD $CLI_CMD" in
-        *$DATA_FILES_PATH/*)
+    # Check if our test programs use files.
+    case "$CLI_CMD" in
+        *programs/ssl/*"$DATA_FILES_PATH/"*)
+            requires_config_enabled MBEDTLS_FS_IO;;
+    esac
+    case "$SRV_CMD" in
+        *programs/ssl/*"$DATA_FILES_PATH/"*)
             requires_config_enabled MBEDTLS_FS_IO;;
     esac
```

Comment:
So this innocent-looking change (at least it looked innocent to me) is causing a bunch more test cases to be executed in `config-suite-b`, `config-ccm-psk-tls1_2.h` and `config-ccm-psk-dtls1_2.h`. And some are failing because a genuine requirement isn't declared. The joys of ssl-opt.

Reply:
And by the way, maybe we should enable `MBEDTLS_FS_IO` in `test-ref-configs` when we run `ssl-opt` (we already add `MBEDTLS_DEBUG_C` and `MBEDTLS_ERROR_C`)?

Reply:
Aw, let's not do that now, then.

> The joys of ssl-opt.

Yeah, it was designed to only work in the default config, and dependency declarations were added later as an after-thought, and for some time the target was it should run in the default and full configs, then perhaps a few others that are close enough... but we never took the time to think it through, and here we are now.

> And by the way, maybe we should enable MBEDTLS_FS_IO in test-ref-configs when we run ssl-opt (we already add MBEDTLS_DEBUG_C and MBEDTLS_ERROR_C)?

Yeah, I think that would make sense. Perhaps for your other PR that splits test-ref-configs? (Unless it turns out to be another can of worms.)